gridt/src/app/core/settings.service.spec.ts

```diff
@@ -93,32 +123,183 @@ describe("SettingsService when authentication fails", () => {
+    secStoreStub.get$.and.returnValue(of(JSON.stringify(mock_id[0])));
+    apiStub.userIdentity$.and.returnValue(of(mock_id[0]));
+    service.updateIdentity();
+    expect(service.userIdentity$).toBeObservable(cold('a',{a: mock_id[0]}));
+  });
+
+  it("should deal with many async calls", () => {
+    let mockIDs = {
+      a: mock_id[0],
+      b: mock_id[1],
+      c: mock_id[2],
+      e: service.empty_identity
+    };
+
+    /**
+     * Tests the async behavior of userIdentity$.
+     * - If both server and local identity are unavailable, emit empty ID.
+     *   [step 1]
+     * - When server ID is retreived, but local ID has not been set yet, store
+     *   server ID in local and emit local ID. [step 2]
+     * - When the server ID is different from local ID, overwrite local ID with
+     *   the server ID. Then emit local ID. [step 3,9]
+     * - If both the server and local identity agree, emit local ID. [step 4]
+     * - If the server is unavailable but local ID is available, emit local ID.
+     *   [step 5, 6]
+     * - Only when updateIdentity() function is called should the userIdentity$
+     *   change. [step 7,8]
+     */
+    let serverID            = cold('--abb-e--c--', mockIDs);
+    let setLocalID          = cold('-------a--b-', mockIDs);
+    let expectedUserID      = cold('e-abb-ba---c', mockIDs);
+    let userIDUpdates       = cold('q-qqq-qq---q', {q: true});
+
+    // Let server and local response be erronous at first
+    apiStub.userIdentity$.and.returnValue(throwError("Server not available"));
+    secStoreStub.get$.and.returnValue(throwError("Local storage not available"));
+
+    serverID.subscribe( (id) => {
+      apiStub.userIdentity$.and.returnValue(of(id));
+    });
+
+    setLocalID.subscribe( (id) => secStoreStub.get$.and.returnValue(
+      of(JSON.stringify(id))
+    ));
+
+    // When secStore is set, make sure it now returns right value upon get.
+    secStoreStub.set$.and.callFake( (key: string, value: any) => {
+      secStoreStub.get$.and.returnValue(of(value));
+      return of(true);
+    });
+
+    userIDUpdates.subscribe( () => service.updateIdentity() );
+
+    expect(service.userIdentity$).toBeObservable(expectedUserID);
+  });
+
+  it("should wait for the server to respond", fakeAsync(() => {
+    // let fake_now = new Date();
+    // fake_now.setUTCHours(1,0,0,0);
+    // jasmine.clock().mockDate(fake_now);
+
+    apiStub.userIdentity$.and.returnValue(
+      timer(30).pipe(
+        take(1),
+        flatMap( () => of(mock_id[1]))
+      )
+    );
+
+    // When secStore is set, make sure it now returns right value upon get.
+    secStoreStub.set$.and.callFake( (key: string, value: any) => {
+      secStoreStub.get$.and.returnValue(of(value));
+      return of(true);
+    });
+
+    service.updateIdentity();
+    tick(15);
+    expect(service.userIdentity$).toBeObservable(cold('',{}));
+    tick(30);
+    expect(service.userIdentity$).toBeObservable(cold('a',{a:mock_id[1]}));
```

Comment:
As far as I understand, changing the local time does not have an effect on the observables, as time is being mocked in it's entirety by jasmine-marbles and the executing of it takes place instantly.

Reply:
I'm not sure what you mean with this comment. Do you mean I could leave out
```
    tick(15);
    expect(service.userIdentity$).toBeObservable(cold('',{}));
    tick(30);
````
?

Reply:
I mean that you could take out the `tick()`s.

Reply:
Yet if I do, the test fails. The idea of this test is to mock the server response with the timer function. This only fires after 30ms. Hence, up to that point the userIdentity$ must remain undefined. After 30ms it should be set. That timing must be done with the tick() functions. Good thing is that Jasmine takes care of the tick() functions and thus indeed evaluates everything at once.

Reply:
Then I probably lack a good understanding of your code. Let me take another look